libmamba/src/api/configuration.cpp

```diff
@@ -490,6 +490,18 @@ namespace mamba
             {
                 for (const auto& dir : dirs)
                 {
+                    if (!fs::exists(dir))
```

Comment:
`create_directories` shouldn't report errors when attempting to create an already existing directory, so you can remove that `if` and call `create_directories` every time (keep the error handling for the other potential errors).